### PR TITLE
GraphQL changes for 2.0

### PR DESCRIPTION
**Pull request by TonyCTHsu**
**2.0 Upgrade Guide notes**

🚨 Breaking changes: 

1. No longer support and patch GraphQL's defined-based schema

2. Support `graphql-ruby` versions containing breaking changes.

* 1.13.21 =>https://github.com/rmosolgo/graphql-ruby/pull/4788
* 2.0.28 => https://github.com/rmosolgo/graphql-ruby/pull/4789
* 2.1.11 => https://github.com/rmosolgo/graphql-ruby/pull/4786
* 2.2.6 => https://github.com/rmosolgo/graphql-ruby/pull/4776

3. Change `schemas` option default

**Before**: Your application's GraphQL schemas have to be explicitly provided through `schemas` option to be instrumented. 

**Now**: GraphQL `schemas` is no longer required to be explicitly provided to `schemas` option. By default, every schema is instrumented.

-----------

**Note:** 1.13.x is only partially support(only class-based schema)
**Note:** Integration supported versions will be updated once `graphql-ruby` releases our patch
**Note:** Doc will be updated once `graphql-ruby` releases our patch

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -88,7 +88,7 @@ TEST_METADATA = {
     'graphql-2.1'  => '❌ 2.1 / ❌ 2.2 / ❌ 2.3 / ❌ 2.4 / ❌ 2.5 / ❌ 2.6 / ❌ 2.7 / ❌ 3.0 / ✅ 3.1 / ❌ 3.2 / ❌ 3.3 / ❌ jruby',
     'graphql-2.0'  => '❌ 2.1 / ❌ 2.2 / ❌ 2.3 / ❌ 2.4 / ❌ 2.5 / ❌ 2.6 / ❌ 2.7 / ❌ 3.0 / ✅ 3.1 / ❌ 3.2 / ❌ 3.3 / ❌ jruby',
     'graphql-1.13' => '❌ 2.1 / ❌ 2.2 / ❌ 2.3 / ❌ 2.4 / ❌ 2.5 / ❌ 2.6 / ❌ 2.7 / ❌ 3.0 / ✅ 3.1 / ❌ 3.2 / ❌ 3.3 / ❌ jruby',
-    'graphql-1.12' => '❌ 2.1 / ❌ 2.2 / ❌ 2.3 / ❌ 2.4 / ❌ 2.5 / ❌ 2.6 / ❌ 2.7 / ❌ 3.0 / ✅ 3.1 / ❌ 3.2 / ❌ 3.3 / ❌ jruby',
+    # 'graphql-1.12' => '❌ 2.1 / ❌ 2.2 / ❌ 2.3 / ❌ 2.4 / ❌ 2.5 / ❌ 2.6 / ❌ 2.7 / ❌ 3.0 / ✅ 3.1 / ❌ 3.2 / ❌ 3.3 / ❌ jruby',
```
Comment (marcotc) [inline Rakefile:91]: We can just remove the line if we are dropping support for graphql 1.12.

**Review by marcotc — APPROVED** [review]

File: Rakefile [hunk 2]
```diff
@@ -88,7 +88,7 @@ TEST_METADATA = {
     'graphql-2.1'  => '❌ 2.1 / ❌ 2.2 / ❌ 2.3 / ❌ 2.4 / ❌ 2.5 / ❌ 2.6 / ❌ 2.7 / ❌ 3.0 / ✅ 3.1 / ❌ 3.2 / ❌ 3.3 / ❌ jruby',
     'graphql-2.0'  => '❌ 2.1 / ❌ 2.2 / ❌ 2.3 / ❌ 2.4 / ❌ 2.5 / ❌ 2.6 / ❌ 2.7 / ❌ 3.0 / ✅ 3.1 / ❌ 3.2 / ❌ 3.3 / ❌ jruby',
     'graphql-1.13' => '❌ 2.1 / ❌ 2.2 / ❌ 2.3 / ❌ 2.4 / ❌ 2.5 / ❌ 2.6 / ❌ 2.7 / ❌ 3.0 / ✅ 3.1 / ❌ 3.2 / ❌ 3.3 / ❌ jruby',
-    'graphql-1.12' => '❌ 2.1 / ❌ 2.2 / ❌ 2.3 / ❌ 2.4 / ❌ 2.5 / ❌ 2.6 / ❌ 2.7 / ❌ 3.0 / ✅ 3.1 / ❌ 3.2 / ❌ 3.3 / ❌ jruby',
```
Comment (ivoanjo) [inline Rakefile:91]: This is surprising -- why can't we test graphql on other Ruby versions? :thinking: :eyes:

Comment (TonyCTHsu) [reply Rakefile:91]: Our 2.0 release contains breaking changes that makes existing graphql-ruby code not functional

Comment (TonyCTHsu) [reply Rakefile:91]: I was using 3.1 as an example and will update the matrix with other Ruby versions once the changes are confirmed. 

Also, the dependency should not be installing from my fork but the official gem release. 😄

Comment (ivoanjo) [reply Rakefile:91]: Got it -- thanks for clarifying :bow:

**Pull request merged**